src/main/java/seedu/trackbeau/model/customer/Customer.java

```diff
@@ -113,7 +113,40 @@ public boolean isSameItem(UniqueListItem other) {
         }
 
         Customer otherCustomer = (Customer) other;
-        return otherCustomer.getName().equals(getName());
+        return hasSameIdentity(otherCustomer);
+    }
+
+    /**
+     * Returns true if edited customer has changed phone or email fields.
+     */
+    public boolean isDifferentItemFromOriginal(UniqueListItem other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Customer)) {
+            return false;
+        }
+
+        Customer otherCustomer = (Customer) other;
+        boolean hasSamePhoneNumber = otherCustomer.getPhone().equals(getPhone());
+        boolean hasSameEmail = otherCustomer.getEmail().equals(getEmail());
+        return !(hasSamePhoneNumber && hasSameEmail);
+    }
+
+    /**
+     * Returns true if two customers have same name or same phone number or same email
+     * @param otherCustomer
+     */
+    boolean hasSameIdentity(Customer otherCustomer) {
+        boolean hasSamePhoneNumber = otherCustomer.getPhone().equals(getPhone());
+        boolean hasSameEmail = otherCustomer.getEmail().equals(getEmail());
+        if (hasSameEmail || hasSamePhoneNumber) {
```

Comment:
I think based on the Javadocs header comment need to check but I think we can just check for the same phone number or same email? Since no 2 person can have the same phone number or email.